yb-voyager/src/query/queryissue/detectors_ddl.go

```diff
@@ -680,6 +680,52 @@ func (d *IndexIssueDetector) DetectIssues(obj queryparser.DDLObject) ([]QueryIss
 					}
 				}
 			}
+			if idx == 0 && !param.IsExpression {
+				//if this is first column and not an expression index
+				indexIssues, err := d.reportVariousIndexPerfOptimizationsOnFirstColumnOfIndex(index)
+				if err != nil {
+					return nil, err
+				}
+				issues = append(issues, indexIssues...)
+			}
+		}
+	}
+	return issues, nil
+}
+
+func (i *IndexIssueDetector) reportVariousIndexPerfOptimizationsOnFirstColumnOfIndex(index *queryparser.Index) ([]QueryIssue, error) {
+	var issues []QueryIssue
+
+	firstColumnParam := index.Params[0]
+	firstColumnName := fmt.Sprintf("%s.%s", index.GetTableName(), firstColumnParam.ColName)
+
+	isSingleColumnIndex := len(index.Params) == 1
+
+	for column, stat := range i.columnStatistics {
+		if firstColumnName != column {
+			continue
+		}
+
+		maxFrequencyPerc := int(stat.MaxFrequency * 100)
+		nullFrequencyPerc := int(stat.NullFrequency * 100)
+
+		if stat.DistinctValues > LOW_CARDINALITY_MIN_THRESHOLD && stat.DistinctValues <= LOW_CARDINALITY_MAX_THRESHOLD {
```

Comment:
can you add a comment here describing the precendence order and the reasoning for that.